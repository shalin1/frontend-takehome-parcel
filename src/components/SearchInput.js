import React from 'react';
import PropTypes from 'prop-types';

const SearchInput = ({onChange,value}) => (
    <div role="search" key="container">
        <label htmlFor="search-input">Search</label>
        <input
            value={value}
            onChange={e=>onChange(e.target.value)}
            placeholder="Search Gems..."
            type="search"
            id="search-input"
            name="search-input"
        />
    </div>
)

SearchInput.propTypes = {
    onChange: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired
}

export default SearchInput