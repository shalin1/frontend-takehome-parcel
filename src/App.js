import React, {useState} from 'react'
import SearchInput from './components/SearchInput'

const App = () => {
    const [searchTerm,setSearchTerm] = useState('')

    return (
        <div role='main' key='app'>
            <h1>✨👋🌍Hello World</h1>
            <SearchInput value={searchTerm} onChange={setSearchTerm}/>
        </div>
    )
}

export default App