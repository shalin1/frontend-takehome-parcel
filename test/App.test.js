import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import {axe, toHaveNoViolations} from "jest-axe";
expect.extend(toHaveNoViolations);
import App from '../src/App';
import SearchInput from '../src/components/SearchInput';

describe('App', () => {
    const app = shallow(<App />);
    it('renders', () => {
        expect(app.exists()).toBe(true)
    })

    it('has not changed from the last snapshot', () => {
        const tree = renderer.create(app).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('has no programmatically detectable a11y issues', async () => {
        const results = await axe(app.html());
        expect(results).toHaveNoViolations();
    })

    it('renders a hello world header', () => {
        expect(app.find('h1').exists()).toBe(true);
        expect(app.text()).toMatch(/Hello World/)
    });

    it('renders a single search input', () => {
        const input = app.find(SearchInput)
        expect(input.length).toBe(1);
    })

    it('passes the correct props to the search input', () => {
        const input = app.find(SearchInput)
        expect(input.props()).toEqual({onChange: expect.any(Function), value: ''})
    })

    it("persists changes to the input's value", () => {
        const wrapper = mount(<App/>)
        wrapper.find('input').simulate('change', { target: { value: 'new value' } })
        expect(wrapper.find('input').props().value).toBe('new value')
    })
});
