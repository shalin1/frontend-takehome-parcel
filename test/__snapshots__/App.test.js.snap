// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has not changed from the last snapshot 1`] = `
<div
  role="main"
>
  <h1>
    ✨👋🌍Hello World
  </h1>
  <div
    role="search"
  >
    <label
      htmlFor="search-input"
    >
      Search
    </label>
    <input
      id="search-input"
      name="search-input"
      onChange={[Function]}
      placeholder="Search Gems..."
      type="search"
      value=""
    />
  </div>
</div>
`;
